docs/permission/writing-a-policy.md

```diff
@@ -3,3 +3,137 @@ id: writing-a-policy
+
+As we confirmed in the previous section, we know that this now prevents us from unregistering catalog components. Hooray! But you may notice that this prevents _anyone_ from unregistering a component, which is not a very realistic policy. Let's improve this policy by disabling the unregister action _unless you are the owner of this component_.
+
+## Conditional decisions
+
+Let's change the policy to the following:
+
+```diff
+- import { IdentityClient } from '@backstage/plugin-auth-node';
++ import { BackstageIdentityResponse, IdentityClient } from '@backstage/plugin-auth-node';
++ import { catalogConditions, createCatalogPolicyDecision } from '@backstage/plugin-catalog-backend';
+
+  ...
+
+  class TestPermissionPolicy implements PermissionPolicy {
+-   async handle(request: PolicyAuthorizeQuery): Promise<PolicyDecision> {
++   async handle(
++     request: PolicyAuthorizeQuery,
++     user?: BackstageIdentityResponse,
++    ): Promise<PolicyDecision> {
+      if (request.permission.name === 'catalog.entity.delete') {
+-       return {
+-         result: AuthorizeResult.DENY,
+-       };
++       return createCatalogPolicyDecision(
++         catalogConditions.isEntityOwner(
++           user?.identity.ownershipEntityRefs ?? [],
++         ),
++       );
+      }
+
+      return { result: AuthorizeResult.ALLOW };
+    }
+  }
+```
+
+Let's walk through the new code that we just added. Inside of the if statement, instead of returning a Definitive Policy Decision of DENY, we now return a [Conditional Policy Decision](https://backstage.io/docs/reference/plugin-permission-node.conditionalpolicydecision) (See the [Concepts page](./concepts.md) for more details). This is a way for policies to defer the evaulation of the decision back to the plugin which owns the permission. This allows the framework to support cases in which the policy does not have all the information required to make a decision.
+
+In the policy above, there's no way for the handle method to determine whether the user who is trying to unregister the entity is the owner of that entity. So we use the `createCatalogPolicyDecision` helper provided by the catalog backend to craft a conditional decision, which allows us to tell the catalog backend that it should only return ALLOW if the user owns the entity.
+
+The `catalogConditions` object contains various conditions that the catalog plug authors have provided for us to use in authoring our policy. Thankfully, they have provided the `isEntityOwner` rule, which is exactly what we need.
```

Comment:
Is it perhaps worth calling out here that it's possible to supply additional conditions to the catalog on top of the built-in ones?

Reply:
Originally when I was writing this, I left this out in order to keep the tutorial focused on basics. But then I realized I don't think we call this out anywhere else, and this seems like the best place out of all the sections/pages we have so far. Will add a snippet!